### PR TITLE
refactor(ui): move overview test results to TestResults component

### PR DESCRIPTION
**Pull request by squidsoup**
## Done

- move overview test results to TestResults component

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Ensure overview cards still render correctly, open appropriate testing modals and link to testing tab.

## Fixes

Fixes: #1837 

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineSummary/TestResults/TestResults.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import { HardwareType } from "app/base/enum";
+
+type Props = {
+  machine: MachineDetails;
+  hardwareType: HardwareType;
+  setSelectedAction: SetSelectedAction;
+};
+
+const getScriptType = (hardwareType: HardwareType) =>
+  HardwareType[hardwareType].toLowerCase();
```
Comment (Caleb-Ellis) [inline ui/src/app/machines/views/MachineDetails/MachineSummary/TestResults/TestResults.tsx:19]: Just thought you should know I spent an embarrassing amount of time trying to parse this thinking `HardwareType` was a normal TS type. :grimacing:

**Review by Caleb-Ellis — COMMENTED** [review]
One suggestion about wording. Also would you mind adding this to the network card as well?

File: ui/src/app/machines/views/MachineDetails/MachineSummary/OverviewCard/CpuCard/CpuCard.test.tsx [hunk 2]
```diff
@@ -176,6 +176,6 @@ describe("CpuCard", () => {
 
     expect(
       wrapper.find("[data-test='tests']").childAt(0).find("Button").text()
-    ).toContain("Test CPU");
+    ).toContain("Test cpu");
```
[the remark this answers is not in the document]
Comment (squidsoup) [reply ui/src/app/machines/views/MachineDetails/MachineSummary/OverviewCard/CpuCard/CpuCard.test.tsx:179]: hmm agreed, I'll add a displayName function of some sort.

**Review by Caleb-Ellis — APPROVED** [review]
LGTM! Thanks for adding it to the network card!

**Pull request merged**